server/store/missions.js

```diff
@@ -7,11 +7,11 @@ const getMission = async missionId => {
   return await redis.hgetallAsync(`missions:${missionId}`);
 };
 
-const getLatestMissionId = async userId => {
-  // use zrevrange to reverse sorted set from highest to lowest
-  // reversed values will put most recent timestamp at the top
+const getLatestMission = async userId => {
   const missions = await redis.zrevrangeAsync(`user_missions:${userId}`, 0, -1);
-  return missions[0];
+  const mission = await getMission(missions[0]);
```

Comment:
Perhaps above this line you want to make sure that `missions` is not empty.
And below it make sure mission is an object.